tools/scylla-sstable-crawler.cc

```diff
@@ -0,0 +1,693 @@
+$ scylla-sstable-crawler dump --merge /path/to/md-123456-big-Data.db /path/to/md-123457-big-Data.db
+
+# generate a joint histogram for the specified partition
+$ scylla-sstable-crawler writetime-histogram --partition={{myhexpartitionkey}} /path/to/md-123456-big-Data.db
+
+# validate the specified sstables
+$ scylla-sstable-crawler validate /path/to/md-123456-big-Data.db /path/to/md-123457-big-Data.db
+)";
+    app_cfg.description = format(description_template, app_name, boost::algorithm::join(operations | boost::adaptors::transformed(
+                    [] (const auto& op) { return format("* {}: {}", op->name(), op->description()); } ), "\n"));
+
+    const auto op_list = boost::algorithm::join(operations | boost::adaptors::transformed(
+                    [] (const auto& op) { return sstring(op->name()); } ), ", ");
+
+    const auto op_help = fmt::format("operation to run, one of ({})", op_list);
+
+    app_template app(std::move(app_cfg));
+
+    app.add_options()
+        ("schema-file", bpo::value<sstring>()->default_value("schema.cql"), "file containing the schema description")
+        ("partition", bpo::value<std::vector<sstring>>(), "partition(s) to filter for, partitions are expected to be in the hex format")
+        ("partitions-file", bpo::value<sstring>(), "file containing partition(s) to filter for, partitions are expected to be in the hex format")
+        ("merge", "merge all sstables into a single mutation fragment stream (use a combining reader over all sstable readers)")
+        ("no-skips", "don't use skips to skip to next partition when the partition filter rejects one, this is slower but works with corrupt index")
+        ;
+
+    app.add_positional_options({
+        {"operation", bpo::value<sstring>()->default_value(sstring(operations.front()->name())), op_help.c_str(), 1},
+        {"sstables", bpo::value<std::vector<sstring>>(), "sstable(s) to process, can also be provided as positional arguments", -1},
+    });
+
+    //FIXME: this exposes all core options, which we are not interested in.
+    return app.run(argc, argv, [&app, &op_list] {
+        return async([&app, &op_list] {
+            auto& app_config = app.configuration();
+
+            const auto op_name = app_config["operation"].as<sstring>();
+            auto it = std::find_if(operations.begin(), operations.end(), [op_name] (const auto& op) {
+                return op->name() == op_name;
+            });
+            if (it == operations.end()) {
+                throw std::invalid_argument(fmt::format("error: invalid operation, expected one of [{}], got {}", op_list, op_name));
+            }
+            const auto& operation = **it;
+
+            operation::options opts;
+            opts.merge = app_config.count("merge");
+            opts.no_skips = app_config.count("no-skips");
+
+            const auto schema = tools::load_schema(std::filesystem::path(app_config["schema-file"].as<sstring>())).get();
```

Comment:
Does it make sense to support also giving the schema as an immediate-value argument to the option?

Reply:
I'm not sure what you mean? What is an "immediate-value argument"?

Reply:
just having an option to specify the schema as a string like `--schema "CREATE TABLE ..."`

Reply:
Ah. The schema definition is usually quite a big string, not really fit for a command line argument. Maybe it can work for well smaller schemas though. This will require support for automatically creating a keyspace if one is not specified. I'll add this to the follow-up TODO list along with the `--schema` argument.

Reply:
Thanks!

Reply:
This doesn't work unfortunately as boost doesn't like string parameter values which have whitespace in them.